analyses/cell-type-consensus/scripts/04-assign-consensus-celltypes.R

```diff
@@ -0,0 +1,171 @@
+  paste0(collapse = ";")
+project_id <- metadata(sce)$project_id
+
+# check if cell line since cell lines don't have any cell type assignments
+# account for having more than one sample and a list of sample types
+# all sample types should be the same theoretically
+sample_type <- unique(metadata(sce)$sample_type)
+is_cell_line <- sample_type == "cell line"
+
+# grab coldata
+coldata_df <- colData(sce) |>
+  as.data.frame() |>
+  # add unique sample/library information
+  dplyr::mutate(
+    project_id = project_id,
+    sample_id = sample_id,
+    library_id = library_id,
+    # add in sample type to make sure we don't assign consensus cell types to cell lines
+    sample_type = sample_type
+  )
+
+# only select sample info and cell type info, we don't need the rest of the coldata
+# if sample is cell line, fill in celltype columns with NA
+if (is_cell_line) {
+  celltype_df <- coldata_df |>
+    dplyr::select(
+      project_id,
+      sample_id,
+      library_id,
+      barcodes,
+      sample_type
+    ) |>
+    dplyr::mutate(
+      singler_celltype_ontology = NA,
+      singler_celltype_annotation = NA,
+      cellassign_celltype_annotation = NA
+    )
+} else {
+  # otherwise select the cell type columns
+  celltype_df <- coldata_df |>
+    dplyr::select(
+      project_id,
+      sample_id,
+      library_id,
+      barcodes,
+      sample_type,
+      contains("celltype") # get both singler and cellassign with ontology
+    )
+}
+
+
+# Prep references --------------------------------------------------------------
+
+# change names for panglao ref to match what's in the consensus file
+panglao_ref_df <- readr::read_tsv(opt$panglao_ref_file) |>
+  dplyr::rename(
+    panglao_ontology = ontology_id,
+    panglao_annotation = human_readable_value,
+    original_panglao_name = panglao_cell_type
+  )
+
+consensus_ref_df <- readr::read_tsv(opt$consensus_ref_file) |>
+  # select columns to use for joining and consensus assigmments
+  dplyr::select(
+    panglao_ontology,
+    original_panglao_name,
+    blueprint_ontology,
+    consensus_annotation,
+    consensus_ontology
+  )
+
+# grab blueprint ref 
+blueprint_df <- readr::read_tsv(opt$blueprint_ref_file)
+
+# Create combined TSV ----------------------------------------------------------
+
+all_assignments_df <- celltype_df |> 
+  # add columns for panglao ontology and consensus
+  # first add panglao ontology
+  dplyr::left_join(panglao_ref_df, by = c("cellassign_celltype_annotation" = "original_panglao_name")) |>
+  # now add in all the blueprint columns
+  dplyr::left_join(blueprint_df, by = c("singler_celltype_ontology" = "blueprint_ontology")) |>
+  # then add consensus labels
+  dplyr::left_join(consensus_ref_df,
+                   by = c(
+                     "singler_celltype_ontology" = "blueprint_ontology",
+                     "cellassign_celltype_annotation" = "original_panglao_name",
+                     "panglao_ontology"
+                   )
+  ) |>
```

Comment:
Just formatting for consistency
```suggestion
  dplyr::left_join(
    consensus_ref_df,
    by = c(
      "singler_celltype_ontology" = "blueprint_ontology",
      "cellassign_celltype_annotation" = "original_panglao_name",
      "panglao_ontology"
    )
  ) |>
```